processing/src/test/java/io/druid/query/topn/TopNQueryRunnerTest.java

```diff
@@ -3678,4 +3679,351 @@ public void testTopNWithExtractionFilterNoExistingValue()
     final Sequence<Result<TopNResultValue>> retval = runWithPreMergeAndMerge(topNQueryWithNULLValueExtraction);
     TestHelper.assertExpectedResults(expectedResults, retval);
   }
+
+  @Test
+  public void testFullOnTopNFloatColumn()
```

Comment:
Similar to groupBy, please include more tests covering:

- Long columns other than `__time`
- DefaultDimensionSpec on a column that is actually string but output type is LONG or FLOAT
- DefaultDimensionSpec on a column that is actually long but output type is STRING